bundles/org.openhab.binding.pegelonline/src/main/java/org/openhab/binding/pegelonline/internal/discovery/PegelDiscovery.java

```diff
@@ -0,0 +1,122 @@
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.config.discovery.DiscoveryService;
+import org.openhab.core.i18n.LocationProvider;
+import org.openhab.core.io.net.http.HttpClientFactory;
+import org.openhab.core.library.types.PointType;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * {@link PegelDiscovery} Discovery of measurement stations
+ *
+ * @author Bernd Weymann - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = DiscoveryService.class, immediate = true, configurationPid = "discovery.pegelonline")
+public class PegelDiscovery extends AbstractDiscoveryService implements DiscoveryService, ThingHandlerService {
+    private final Logger logger = LoggerFactory.getLogger(PegelDiscovery.class);
+    private Optional<PegelOnlineHandler> handler = Optional.empty();
+    private PointType homeLocation = UNDEF_LOCATION;
+    private HttpClientFactory httpClientFactory;
+
+    @Activate
+    public PegelDiscovery(final @Reference HttpClientFactory hcf, final @Reference LocationProvider lp) {
+        super(SUPPORTED_THING_TYPES_UIDS, 10, false);
+        httpClientFactory = hcf;
+        PointType location = lp.getLocation();
+        if (location != null) {
+            homeLocation = location;
+        } else {
+            logger.debug("No home location found");
+        }
+    }
+
+    @Override
+    protected void startScan() {
+        double homeLat = homeLocation.getLatitude().doubleValue();
+        double homeLon = homeLocation.getLongitude().doubleValue();
+        try {
+            ContentResponse cr = httpClientFactory.getCommonHttpClient().GET(STATIONS_URI);
+            Station[] stationArray = GSON.fromJson(cr.getContentAsString(), Station[].class);
+            if (stationArray != null) {
+                for (Station s : stationArray) {
+                    double distance = Utils.getDistanceFromLatLonInKm(homeLat, homeLon, s.latitude, s.longitude);
+                    if (distance < DISCOVERY_RADIUS) {
+                        logger.trace("Station in range {},{}", s.longname, s.water.shortname);
+                        reportResult(s);
+                    }
+                }
+            } else {
+                logger.trace("No stations found in discovery");
+            }
+        } catch (ExecutionException | TimeoutException | InterruptedException e) {
+            logger.trace("Exception during station discovery: {}", e.getMessage());
+        }
+    }
+
+    public void reportResult(Station s) {
+        String label = "Pegel Station " + Utils.toTitleCase(s.shortname) + " / " + Utils.toTitleCase(s.water.shortname);
```

Comment:
I think you can use this method:
https://www.openhab.org/javadoc/latest/org/openhab/core/util/stringutils#capitalize(java.lang.String)

Reply:
imho `Utils.toTitleCase` can be removed at all.

Reply:
reworked